Prepare the metrics reference for Cloud users

ptgott opened the pull request:
Arrange metrics into H2 sections, both making the page easier to
navigate and making it clear which metrics are relevant to Cloud
users.
    
Add a warning that in Cloud, the Auth and Proxy do not expose
metrics endpoints.

See #10633 

The first commit in this PR tries an alternative approach using Tabs,
which I thought was messier than this approach.

ptgott committed:
Metrics guide

Add separate Tabs for self-hosted and Cloud editions

ptgott committed:
Prepare the metrics reference for Cloud users

Arrange metrics into H2 sections, both making the page easier to
navigate and making it clear which metrics are relevant to Cloud
users.

Add a warning that in Cloud, the Auth and Proxy do not expose
metrics endpoints.

zmb3 commented on docs/pages/setup/reference/metrics.mdx:
```diff
@@ -54,70 +60,98 @@ Now you can see the monitoring information by visiting several endpoints:
 | `firestore_events_backend_batch_read_seconds` | histogram | GCP Cloud Firestore | Latency for Cloud Firestore events batch read operations. |
 | `firestore_events_backend_batch_write_requests` | counter | GCP Cloud Firestore | Number of batch write requests to Cloud Firestore events. |
 | `firestore_events_backend_batch_write_seconds` | histogram | GCP Cloud Firestore | Latency for Cloud Firestore events batch write operations. |
+| `gcs_event_storage_downloads_seconds` | histogram | Internal Golang | Latency for GCS download operations. |
```
The `gcs_` metrics are not `Internal Golang` - they're from the GCS (Google Cloud Storage) session uploader.

ptgott replied:
Oversight on my part. Thanks for catching.

zmb3 commented on docs/pages/setup/reference/metrics.mdx:
```diff
@@ -44,7 +51,6 @@ Now you can see the monitoring information by visiting several endpoints:
 | `backend_read_seconds` | histogram | cache | Latency for read operations. |
 | `backend_write_requests_total` | counter | cache | Number of write requests to the backend. |
 | `backend_write_seconds` | histogram | cache | Latency for backend write operations. |
-| `certificate_mismatch_total` | counter | Teleport Proxy | Number of times there was a certificate mismatch. |
```
Why was this removed?

In the source, the description is `counts login failures due to certificate mismatch`

ptgott replied:
I moved this to the H2 section related to the Proxy. Looking at the source, though, this is [incremented](https://github.com/gravitational/teleport/blob/c0205a737108c0a43b15f0781bf234992952b9e9/lib/srv/authhandlers.go#L288) in `*AuthHandlers.UserKeyAuth`, which seems to be used when initializing SSH servers in both Nodes and the Proxy. I've suggested adding this to the "All Teleport Instances" section.

zmb3 commented on docs/pages/setup/reference/metrics.mdx:
```diff
@@ -54,70 +60,98 @@ Now you can see the monitoring information by visiting several endpoints:
+| Name | Type | Component | Description |
+| - | - | - | - |
 | `reversetunnel_connected_proxies` | gauge | Teleport | Number of known proxies being sought. |
 | `rx` | counter | Teleport | Number of bytes received. |
```
I think `rx` and `tx` might be specific to SSH and could use a better description.

ptgott committed:
Respond to PR feedback

- Move the certificate_mismatch_total to a more appropriate place
  with a more accurate description
- Correct gcs_ metric categories
- Make the rx and tx metric descriptions a bit more accurate
- Also perform light copy-editing on metric descriptions

ulysseskan reviewed (approved):
Updates look good, page is well laid out.  Glad the note is in there about proxy and auth metrics not being available for cloud users.
